src/johnhandler.cpp

```diff
@@ -96,6 +96,9 @@ void JohnHandler::setJohnProgram(const QString &johnProgram)
 
 void JohnHandler::start()
 {
+    // Give a chance to terminate cleanly
+    if (m_john.state() != QProcess::NotRunning)
+        terminate();
```

Comment:
I think we should also use kill = true; Concurrency can bite us in this case so we'd be better of terminating what we don't need.